packages/lib/src/components/MBWay/components/MBWayInput/MBWayInput.tsx

```diff
@@ -57,90 +26,49 @@ function MBWayInput(props: MBWayInputProps) {
     };
 
     this.showValidation = (): void => {
-        setErrors({ ...errors, email: !getIsValidEmail(), phoneNumber: !getIsValidPhoneNumber() });
+        const hasError = !validator.validate('telephoneNumber', 'input')(props.data.telephoneNumber).isValid;
+        setErrors({ ...errors, telephoneNumber: hasError });
     };
 
-    const handleEventFor = (key, type): RtnTypeFnWithEvent => (e: Event): void => {
-        e.preventDefault();
-
+    const handleEventFor = (key: string, mode: string) => (e: Event): void => {
         const val: string = (e.target as HTMLInputElement).value;
+        const { value, isValid }: ValidationObj = validator.validate('telephoneNumber', mode)(val);
 
-        const { value, isValid }: ValidationObj = validator.validate(key, type)(val);
-
-        if (type === 'input') {
-            setData({ ...data, [key]: value });
-            setErrors({ ...errors, [key]: false });
-        }
-
-        if (type === 'blur') {
```

Comment:
By removing this the following scenario is again possible:
- the phone number has less than 7 digits
- it shows an error
- you delete all the digits and click away from the field
- it continues to show the error until 7 digits have been entered
Is this deliberate? We don't normally consider an empty field to be in error until the pay button is pressed

Reply:
That's a good point. I went and tried some components and we don't have a consistent behaviour for this.